File: csharp/yelp-ai-assistant/Routing/QueryRouter.cs
using YelpAiAssistant.Models;

namespace YelpAiAssistant.Routing;

public class QueryRouter
{
    public RoutingDecision Decide(QueryIntent intent) => intent switch
    {
        QueryIntent.Operational => new(intent, true,  false, false),
        QueryIntent.Amenity     => new(intent, true,  true,  true),
        QueryIntent.Quality     => new(intent, false, true,  false),
        QueryIntent.Photo       => new(intent, false, false, true),
        _                       => new(intent, true,  true,  false),
    };
}

File: java/yelp-ai-assistant/src/main/java/com/yelp/assistant/routing/QueryRouter.java
package com.yelp.assistant.routing;

import com.yelp.assistant.model.QueryIntent;
import com.yelp.assistant.model.RoutingDecision;
import org.springframework.stereotype.Component;

@Component
public class QueryRouter {

    public RoutingDecision decide(QueryIntent intent) {
        return switch (intent) {
            case OPERATIONAL -> new RoutingDecision(intent, true, false, false);
            case AMENITY     -> new RoutingDecision(intent, true, true, true);
            case QUALITY     -> new RoutingDecision(intent, false, true, false);
            case PHOTO       -> new RoutingDecision(intent, false, false, true);
            default          -> new RoutingDecision(intent, true, true, false);
        };
    }
}
